extensions/wikia/Chat2/ChatEntryPoint.class.php

```diff
@@ -23,35 +23,56 @@ static public function onParserFirstCallInit( Parser &$parser ) {
 	}
 
 	/**
-	 * @param $isEntryPoint - set to true for chat entry point embeded in the article, false for rail module
-	 * Return an array of variables needed to render chat entry point template
+	 * Return an array of variables needed to render chat entry point mustache template
+	 *
+	 * @param $fromParserTag - set to true for chat entry point embedded in the article,
+	 * set to false for rail module
 	 *
 	 * @return array
 	 */
-	static public function getEntryPointTemplateVars( $isEntryPoint ) {
-		global $wgEnableWallExt, $wgBlankImgUrl;
-		return [
-			'linkToSpecialChat' => SpecialPage::getTitleFor( "Chat" )->escapeLocalUrl(),
-			'isEntryPoint' => $isEntryPoint,
+	static public function getEntryPointTemplateVars( $fromParserTag ) {
```

Comment:
After giving it another thought I think we could make the parameter something like `$mode` introduce two class constants `PARSER_TAG` and `RIGHT_RAIL_MODULE` and rename the template var to be `isParserTag` and add `isRightRailModule` if needed. Sounds reasonable?